src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import {
+  Content,
+  FiatRow,
+  Aligner,
+  InputRow,
+  LabelRow,
+  Container,
+  InputPanel,
+  CurrencySelect,
+  StyledDropDown,
+  StyledTokenName,
+  UppercaseHelper,
+} from './CurrencyInputPanel.styles'
+import Loader from '../Loader'
+import { TYPE } from '../../theme'
+import { RowBetween } from '../Row'
+import DoubleCurrencyLogo from '../DoubleLogo'
+import NumericalInput from '../Input/NumericalInput'
+import { FiatValueDetails } from '../FiatValueDetails'
+import { CurrencyLogo, CurrencyWrapperSource } from '../CurrencyLogo'
+import { CurrencySearchModalComponent } from '../SearchModal/CurrencySearchModal'
+
+import { useActiveWeb3React } from '../../hooks'
+import { useCurrencyBalance } from '../../state/wallet/hooks'
+import { limitNumberOfDecimalPlaces } from '../../utils/prices'
+
+import { CurrencyInputPanelProps } from './CurrencyInputPanel.types'
+
```

Comment:
Recommend to put the whole CurrencyInputPanelProps here, so its easy to check the type definition 

interface CurrencyInputPanelProps {
  id: string
  pair?: Pair | null
  onMax?: () => void
  value: string
  label?: string
  balance?: CurrencyAmount
  currency?: Currency | null
  disabled?: boolean
  hideInput?: boolean
  fiatValue?: CurrencyAmount | null
  isLoading?: boolean
  onUserInput: (value: string) => void
  hideBalance?: boolean
  priceImpact?: Percent
  otherCurrency?: Currency | null
  chainIdOverride?: ChainId
  showCommonBases?: boolean
  onCurrencySelect?: (currency: Currency) => void
  customBalanceText?: string
  isFallbackFiatValue?: boolean
  disableCurrencySelect?: boolean
  currencyWrapperSource?: CurrencyWrapperSource
  displayedValue?: string
  setDisplayedValue?: (val: string) => void
}

Reply:
Imo keeping types separately helps to keep files shorter, also most of editors nowdays have easy way to access type reference. Of course - it's a personal preference.